spring-boot-project/spring-boot-docs/src/main/asciidoc/spring-boot-features.adoc

```diff
@@ -4228,72 +4228,90 @@ the Mongo instance's configuration and logging routing.
 [[boot-features-neo4j]]
 === Neo4j
 http://neo4j.com/[Neo4j] is an open-source NoSQL graph database that uses a rich data
-model of nodes related by first class relationships, which is better suited for connected
-big data than traditional rdbms approaches. Spring Boot offers several conveniences for
+model of nodes connected by first class relationships, which is better suited for connected
+big data than traditional RDBMS approaches. Spring Boot offers several conveniences for
 working with Neo4j, including the `spring-boot-starter-data-neo4j` "`Starter`".
 
 
-
 [[boot-features-connecting-to-neo4j]]
 ==== Connecting to a Neo4j Database
-You can inject an auto-configured `Neo4jSession`, `Session`, or `Neo4jOperations`
-instance as you would any other Spring Bean. By default, the instance tries to connect to
-a Neo4j server at `localhost:7474`. The following example shows how to inject a Neo4j
-bean:
+To access a Neo4j server, you can inject an auto-configured `org.neo4j.ogm.session.Session`. 
+By default, the instance tries to connect to a Neo4j server at `localhost:7687` via the Bolt 
+protocol. The following example shows how to inject a Neo4j session:
 
 [source,java,indent=0]
 ----
 	@Component
 	public class MyBean {
 
-		private final Neo4jTemplate neo4jTemplate;
+		private final Session session;
 
 		@Autowired
-		public MyBean(Neo4jTemplate neo4jTemplate) {
-			this.neo4jTemplate = neo4jTemplate;
+		public MyBean(Session session) {
+			this.session = session;
 		}
 
 		// ...
 
 	}
 ----
 
-You can take full control of the configuration by adding a
-`org.neo4j.ogm.config.Configuration` `@Bean` of your own. Also, adding a `@Bean` of type
-`Neo4jOperations` disables the auto-configuration.
-
-You can configure the user and credentials to use by setting the `spring.data.neo4j.*`
+You can configure the uri and credentials to use by setting the `spring.data.neo4j.*`
 properties, as shown in the following example:
 
 [source,properties,indent=0]
 ----
-	spring.data.neo4j.uri=http://my-server:7474
+	spring.data.neo4j.uri=bolt://my-server:7687
 	spring.data.neo4j.username=neo4j
 	spring.data.neo4j.password=secret
 ----
 
+You can take full control over the session creation by adding a 
+`org.neo4j.ogm.config.Configuration` `@Bean`. To disable the auto-configuration
+provided by the "`Starter`" completely, add a `@Bean` of type 
+`org.neo4j.ogm.session.SessionFactory`. Take note that you have to provide a
```

Comment:
Is the `Neo4jTransactionManager` mandatory in case of custom setup? I am not sure that note should be added here: if users are taking full control, they are responsible to know what needs to be done and refer to the documentation of the components they are configuring themselves.

Reply:
Yes it is, we like to have it mentioned here. As much as I agree with you regarding responsibility, it helps to keep confusion away.

Reply:
I understand but that's not very consistent. If we had to document things that should be done when our auto-config backs off, there are a gazillion of things that should be added to the current ref guide.

Can we take a step back and discuss the source of the confusion? If it's mandatory, perhaps the Neo4j integration could fail with an explicit error message?